vc/vcjwt.go

```diff
@@ -0,0 +1,146 @@
+package vc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/tbd54566975/web5-go/jwt"
+)
+
+// Verify verifies the decoded vc-jwt. It checks for the presence of required fields and verifies the jwt.
+// It returns the decoded vc-jwt and an error if the verification fails.
+func Verify[T CredentialSubject](vcJWT string) (DecodedVCJWT[T], error) {
+	decoded, err := Decode[T](vcJWT)
+	if err != nil {
+		return decoded, err
+	}
+
+	return decoded, decoded.Verify()
+}
+
+// Decode decodes a vc-jwt as per the [spec] and returns [Decoded].
+// Note: this function uses certain fields from the jwt claims to eagrly populate the vc model as described
+// in the encoding section of the spec. The jwt fields will clobber any values that exist in the vc model.
```

Comment:
```suggestion
// Decode decodes a vc-jwt as per the [spec] and returns [Decoded].
// 
// # Note 
// 
// This function uses certain fields from the jwt claims to eagrly populate the vc model as described
// in the encoding section of the spec. The jwt fields will clobber any values that exist in the vc model.
```

nit but I liked your `# Note` pattern in `vc.go`